src/components/shared/map/MapCenter.tsx

```diff
@@ -26,36 +25,37 @@ const crosshairIcon = new L.Icon({
   iconAnchor: [40, 40],
 });
 
-const MapCenter = () => {
-  const t = useTranslations();
+// Define the pin icon for the saved location
+const pinIcon = new L.Icon({
+  iconUrl: '/images/icons/map_of_pi_logo.jpeg',
+  iconSize: [32, 32],
+  iconAnchor: [16, 32],
+});
 
-  const [showPopup, setShowPopup] = useState(false);
-  const [center, setCenter] = useState<{ lat: number; lng: number }>({ lat: 50.064192, lng: 19.944544 });
-  const { currentUser } = useContext(AppContext); // Get currentUser from context
+const MapCenter = () => {
+  const t = useTranslations(); // Hook for internationalization
+  const [showPopup, setShowPopup] = useState(false); // State for controlling the visibility of the confirmation popup
+  const [center, setCenter] = useState<{ lat: number; lng: number }>({ lat: 50.064192, lng: 19.944544 }); // State for storing the map center coordinates
+  const { currentUser } = useContext(AppContext); // Get the current user from context
 
+  // Effect to handle user authentication state
   useEffect(() => {
-    const getMapCenter = async () => {
```

Comment:
How are you planning to fetch the map center after it's persisted?

Reply:
To clarify, the fetchMapCenter function on the frontend makes an HTTP GET request to the /map-center endpoint. This request is handled by the getMapCenter function on the backend, which retrieves and returns the user's saved map center coordinates.

Let me know if there's anything else you'd like to discuss about this approach!